xmppserver/src/main/java/org/jivesoftware/openfire/net/SocketReader.java

```diff
@@ -369,7 +369,7 @@ private IQ getIQ(Element doc) {
             return new Roster(doc);
         }else if (query != null && "jabber:iq:version".equals(query.getNamespaceURI())) {
             IQ iq = new IQ(doc);
-            if (iq.getType().equals(IQ.Type.result) && !iq.getFrom().toString().contains("@") && iq.getFrom().equals(session.getAddress())){
```

Comment:
I think what I did here could be the source of the problem. but I'm not very sure because it was working well with my local Openfire instances.

Reply:
To do better, I can look for another place to put this code as I did with the client response.